### PR TITLE
Code quality

### PR DESCRIPTION
**Pull request by uuunk**
Added rubocop and bundle audit to check for security checks. All these checks will run on CI but I'll recommend to the team that they run `rails check` before committing. Tried using a "overcommit" gem to set them up to run automatically before commits but it didn't play nice with rubymine. If someone wants to manually set their own git precommit hook that's easy enough to do.
https://trello.com/c/k4tYTEOx/67-linter-syntax-checking

Questions for reviewers:
- I set up the rake task to autofix what it can by default. Seems like the easiest route, but let me know if you have concerns with that as the default.
- I did a quick scan of best practices and what others thought of the defaults and changed a few of them. Take a look at the `.rubocop.yml` file and let me know if you have any strong opinions on my overrides or the lack there of.

Lastly I'd like to get this in relatively quick because there are style changes that span so many files. I don't want everyone working on stuff and then having a merge nightmare. If there's a bunch of other PRs out by the time we're ready to push, I'll hold off and clean up when there isn't.

### CODE REVIEW

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -1,35 +1,36 @@
+# frozen_string_literal: true
```
Comment (sanghun89) [inline app/controllers/users_controller.rb]: Just curious - what purpose does this serve???

File: app/controllers/users_controller.rb [hunk 2]
```diff
@@ -1,35 +1,36 @@
...
   end
 
-  def get_donations_by_causes
+  def donations_by_causes
     @donations_by_causes = @user.donations_by_causes
   end
 
   def cause_logos
     {
-      "animals"        => "paw",
```
Comment (markdamobi) [inline app/controllers/users_controller.rb]: I guess rubocop doesn't appreciate readability. @uuunk what's the rule against this?

File: app/controllers/users_controller.rb (hunk 1 above)
Comment (markdamobi) [reply app/controllers/users_controller.rb]: Looks like it's on a lot of files, but not every file. What's the criteria for adding this to a file?

File: app/controllers/users_controller.rb (hunk 2 above)
Comment (markdamobi) [reply app/controllers/users_controller.rb]: There's a migration where I do something similar, but that wasn't flagged. I don't quite understand what's going on here.

**Comment by markdamobi** [comment]
is there something for `erb` files?

File: app/controllers/users_controller.rb (hunk 2 above)
Comment (sanghun89) [reply app/controllers/users_controller.rb]: He's extending off of the default config, which is here https://github.com/rubocop-hq/rubocop/blob/master/config/default.yml

Comment (markdamobi) [reply app/controllers/users_controller.rb]: do you know the exact rule it's using to get rid of that space? I'm just wondering why some things are flagged, but others aren't.

File: app/controllers/users_controller.rb (hunk 1 above)
Comment (uuunk) [reply app/controllers/users_controller.rb]: @sanghun89 It the equivalent of putting "mystring".freeze on every string in the file. It makes them all immutable, kinda like a const in JS. It also saves about 20% on the garbage collection and is going to be default on in Ruby 3.0. Kinda minor but it's recommended for performance and safety. An @markdamobi it'll get automatically put on any run time ruby file when rubocop runs.

File: app/controllers/users_controller.rb (hunk 2 above)
Comment (uuunk) [reply app/controllers/users_controller.rb]: Hmm no I couldn't find anything that would keep that whitespace. Maybe missing. If we come across it I'd be happy to turn it off to make that kind of stuff more readable. I think it sees it as arbitrary whitespace.

**Comment by uuunk** [comment]
> I see the `.rb` files included in the `.rubocop.yml`. Is there a reason for not including `.erb` files?

No that's a good catch, I missed that. I'll add it.

**Pull request merged**